site/docs/latest/advanced/Encryption.md

```diff
@@ -0,0 +1,157 @@
+            producer.send("my-message".getBytes());
+        }
+
+        pulsarClient.close();
+```
+7. Sample Consumer Application:
+```java
+        class EncKeyReader implements CryptoKeyReader {
+
+            Map<String, EncryptionKeyInfo> encKeys = new HashMap<String, EncryptionKeyInfo>();
+
+            EncKeyReader(String keyName, byte[] pubKey, byte[] privKey) {
+
+                EncryptionKeyInfo pubKeyInfo = new EncryptionKeyInfo(pubKey, null);
+                encKeys.put(keyName + ".public-key", pubKeyInfo);
+
+                EncryptionKeyInfo privKeyInfo = new EncryptionKeyInfo(privKey, null);
+                encKeys.put(keyName + ".private-key", privKeyInfo);
+
+            }
+
+            @Override
+            public EncryptionKeyInfo getPublicKey(String keyName, Map<String, String> keyMeta) {
+                return encKeys.get(keyName + ".public-key");
+            }
+
+            @Override
+            public EncryptionKeyInfo getPrivateKey(String keyName, Map<String, String> keyMeta) {
+                return encKeys.get(keyName + ".private-key");
+            }
+        }
+
+        ConsumerConfiguration consConf = new ConsumerConfiguration();
+        consConf.setCryptoKeyReader(
+                new EncKeyReader("myappkey", "public key value".getBytes(), "private key value".getBytes()));
+
+        PulsarClient pulsarClient = PulsarClient.create("http://localhost:8080");
+
+        Consumer consumer = pulsarClient.subscribe("persistent://my-property/use/my-ns/my-topic", "my-subscriber-name", consConf);
+        Message msg = null;
+
+        for (int i = 0; i < 10; i++) {
+            msg = consumer.receive();
+            // do something
+            System.out.println("Received: " + new String(msg.getData()));
+        }
+
+        // Acknowledge the consumption of all messages at once
+        consumer.acknowledge(msg);
+        pulsarClient.close();
+```
+## Key rotation
+Pulsar generates new AES data key every 4 hours or after a certain number of messages are published. The asymmetric public key is automatically fetched by producer every 4 hours to retrieve the latest version.
+
+## Enabling encryption at the producer application:
+If you produce messages that are consumed across application boundaries, you need to ensure that consumers in other applications have access to one of the private keys that can decrypt the messages.  This can be done in two ways:
+1. The consumer application provides you access to their public key, which you add to your producer keys
+1. You grant access to one of the private keys from the pairs used by producer 
+
+In some cases, the producer may want to encrypt the messages with multiple keys. For this, add all such keys to the config. Consumer will be able to decrypt the message, as long as it has access to at least one of the keys.
+
+E.g: If messages needs to be encrypted using 2 keys myapp.messagekey1 and myapp.messagekey2,
+
+conf.addEncryptionKey(“myapp.messagekey1”)<br>
```

Comment:
Use \`\`\` to separated example blocks and remove the `<br>`
Also, example code is missing `;`